### ISSUE TITLE
Possibility to add warning when running yarn in directory containing node_modules but missing yarn.lock

### ISSUE THREAD

**Issue by martinek** [yarnpkg/yarn#3373]
Would it be possible to add some kind of warning when running yarn inside directory where `node_modules` already exist, but there is no `yarn.lock` file?

The only use-case for this should be when migrating from npm to yarn. I realized yarn updated all my `node_modules` only after I ran `yarn` command and then it was too late to search for `yarn import`.

It would be nice to get warning, that running `yarn` or `yarn install` without `yarn.lock` will result in `node_modules` being downloaded and updated.

I realize this experience is depending on environment, but the destructive nature of such simple mistake could warrant some kind of protection against it.

Alternatively, running `yarn` command in this specific case, could by default fallback to `yarn import` command instead of `yarn install`.

Also `yarn import` command could be mentioned on https://yarnpkg.com/lang/en/docs/migrating-from-npm/ page.

**Comment by bestander** [comment]
That sounds like a nice idea.
Send a PR.

**Comment by rubenazevedoff** [comment]
So, whenever you're running yarn, but there is a node_modules in their but no yarn.lock, should give a warning?
Tomorrow morning i will PR this :)

**Comment by bestander** [comment]
If there is a node_modules then there should be a package.json so it should
be fine.

**Comment by rubenazevedoff** [comment]
But this can only happen when you're making the import  right?

**Comment by bestander** [comment]
Yeah or if you don't care that preserve existing node_modules and just
start using yarn.

**Comment by jscoobyced** [comment]
Was this PR submitted? Should the warning be blocking? or should it prompt to "Continue? [Y/n]" ?

**Comment by arcanis** [comment]
No PR has been submitted as far as I know. The warning should not be blocking, we don't want to prevent people from using Yarn :) I think a prompt defaulting to Yes would be fine.

**Comment by msramalho** [comment]
Me and a group of two more are going to address this issue over the next week :smile:
